packages/aws-cdk-lib/custom-resources/test/custom-resource-config/custom-resource-config.test.ts

```diff
@@ -0,0 +1,150 @@
+
+    let websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployNone', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.hasResourceProperties('AWS::Lambda::Function', {
+      LoggingConfig: {
+        LogGroup: {
+          Ref: Match.stringLikeRegexp('CustomCDKBucketDeployment'),
+        },
+      },
+    });
+    template.resourceCountIs('AWS::Logs::LogGroup', 1);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: customResourceLogRetention,
+    });
+  });
+
+  test('addLogRetentionLifetime only modifies custom resource log groups', () => {
+    const customResourceLogRetention = logs.RetentionDays.TEN_YEARS;
+    const nonCustomResourceLogRetention = logs.RetentionDays.TWO_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(customResourceLogRetention);
+    const stack = new cdk.Stack(app);
+
+    const ignored = new logs.LogGroup(stack, 'ignored', {});
+    let websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployNone', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.resourceCountIs('AWS::Logs::LogGroup', 2);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: customResourceLogRetention,
+    });
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: nonCustomResourceLogRetention,
+    });
+  });
+});
+
+describe('when logRetention is specified', () => {
+  test('addLogRetentionLifetime overrides log retention', () => {
+    const customResourceLogRetention = logs.RetentionDays.TEN_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(customResourceLogRetention);
+    const stack = new cdk.Stack(app);
+
+    const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployLogRetention', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logRetention: logs.RetentionDays.ONE_WEEK,
+    });
+    const template = Template.fromStack(stack);
+
+    template.hasResourceProperties('Custom::LogRetention', {
+      RetentionInDays: customResourceLogRetention,
+    });
+    template.resourceCountIs('AWS::Logs::LogGroup', 1);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: customResourceLogRetention,
+    });
+  });
+});
+
+describe('when logGroup is specified', () => {
+  test('when CustomResourceConfig.addLogRetentionLifetime is called, singleton-backed custom resource logGroup period modified', () => {
```

Comment:
"when <something> is called" adds 3 words you don't need. You can improve clarity by always preferring:

"<something> <did something, or performed some action>"

Eg:

```suggestion
  test('addLogRetentionLifetime modifies the retention period of a singleton-backed custom resource log group.', () => {
```

This applies to each test.